appbuilder/services/npm-service.ts

```diff
@@ -75,6 +79,54 @@ export class NpmService implements INpmService {
 		}).future<void>()();
 	}
 
+	public search(projectDir: string, keywords: string[], args?: string[]): IFuture<IBasicPluginInformation[]> {
+		return ((): IBasicPluginInformation[] => {
+			let result: IBasicPluginInformation[] = [];
+			let spawnResult = this.executeNpmCommand(projectDir, args || [], keywords.join(" ")).wait();
+			if (spawnResult.stderr) {
+				// npm will write "npm WARN Building the local index for the first time, please be patient" to the stderr and if it is the only message on the stderr we should ignore it.
+				let splitError = spawnResult.stderr.split("\n");
+				if (splitError.length > 2 || splitError[0].indexOf("Building the local index for the first time") === -1) {
+					this.$errors.failWithoutHelp(spawnResult.stderr);
+				}
+			}
+
+			// Need to split the result only by \n because the npm result contains only \n and on Windows it will not split correctly when using EOL.
+			// Sample output:
+			// NAME                    DESCRIPTION             AUTHOR        DATE       VERSION  KEYWORDS
+			// cordova-plugin-console  Cordova Console Plugin  =csantanapr…  2016-04-20 1.0.3    cordova console ecosystem:cordova cordova-ios
+			let pluginsRows: string[] = spawnResult.stdout.split("\n");
+
+			// Remove the table headers row.
+			pluginsRows.shift();
+
+			let npmNameGroup = "(\\S+)";
+			let npmDateGroup = "(\\d+\\-\\d+\\-\\d+)\\s";
+			let npmFreeTextGroup = "([^=]+)";
+			let npmAuthorsGroup = "((?:=\\S+\\s?)+)\\s+";
+
+			// Should look like this /(\S+)\s+([^=]+)((?:=\S+\s?)+)\s+(\d+\-\d+\-\d+)\s(\S+)(\s+([^=]+))?/
+			let pluginRowRegExp = new RegExp(`${npmNameGroup}\\s+${npmFreeTextGroup}${npmAuthorsGroup}${npmDateGroup}${npmNameGroup}(\\s+${npmFreeTextGroup})?`);
```

Comment:
This whole RegExp can be static also along with all of its components so that they're not created every time the function is executed